bin/oneoff/backfill_data/backfill_script_ids_channel_tokens.rb

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env ruby
+# This script backfills existing ChannelTokens to set script_id
+#
+# Background: previously, the channel token for a level was identified by the level_id column. However,
+# levels are shared across scripts, so we need to identify the channel token by both level_id and script_id.
+# A script_id column was added the the channel tokens table here: https://github.com/code-dot-org/code-dot-org/pull/39835
+# New channel tokens that are generated, are generated with a script_id (https://github.com/code-dot-org/code-dot-org/pull/39855).
+# The purpose of this script is to backfill the script_id for channel tokens created earlier.
+#
+# How this script works:
+# This script will iterate over channel tokens and identify the script_ID in several possible ways,
+# 1. If the level only exists in 1 script, use that script_id
+# 2. If an associated user exists, if the user has progress in one script that the level could be
+# associated with, use that script_id
+# 3. If an associated user exists, if the user has one user level that is associated with the channel
+# token level, use the script_id from the user level
+# 4. If an associated user exists but the user has no script progress or user levels possibly associated
+# with the channel token, it's likely that the user visited the level page, which generated a channel token,
+# and then left the page. In this case, assign any of the possible scripts to the channel token
+
```

Comment:
Wondering if we should add a note here about the common cases of when a script_id will not be filled in (e.g. user was signed out, etc.).